src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsPagesTest.java

```diff
@@ -78,8 +87,13 @@
 import com.puppycrawl.tools.checkstyle.internal.utils.TestUtil;
 import com.puppycrawl.tools.checkstyle.internal.utils.XdocUtil;
 import com.puppycrawl.tools.checkstyle.internal.utils.XmlUtil;
+import com.puppycrawl.tools.checkstyle.site.XdocsTemplateParser;
+import com.puppycrawl.tools.checkstyle.site.XdocsTemplateSinkFactory;
+import com.puppycrawl.tools.checkstyle.utils.CommonUtil;
 import com.puppycrawl.tools.checkstyle.utils.TokenUtil;
 
+// -@cs[ClassDataAbstractionCoupling] No way to split up class usage.
+// Generation should be before validation.
```

Comment:
Can we move all logic to separate class and run it from this test at method that has BeforeAll.
We do not split test as they related to testing module.
This test class is not a test :)

Reply:
Moved to `XdocGenerator`